src/main/webapp/3Dscript.js

```diff
@@ -176,11 +239,93 @@ function plotOrientation() {
   orientation.rotation.y = THREE.Math.degToRad(poseTransform.rotate);
 }
 
+/**
+* get time in readable units
+* @param {double} time raw time up to 6 decimal places. from GPS time
+* @return {String} readable time.
+ */
+function formatTime(time) {
+  /* Time changed from gps to unix. */
+  const gpsDiff = 315964800;
+  const unixTimestamp = time - gpsDiff;
+  // Create a new JavaScript Date object based on the timestamp
+  // multiplied by 1000 so that the argument is in milliseconds, not seconds.
+  const date = new Date(unixTimestamp * 1000);
+  // Hours part from the timestamp
+  const hours = date.getHours();
+  // Minutes part from the timestamp
+  const minutes = '0' + date.getMinutes();
+  // Seconds part from the timestamp
+  const seconds = '0' + date.getSeconds();
+
+  // Will display time in 10:30:23 format
+  const formattedTime = hours + ':' +
+   minutes.substr(-2) + ':' + seconds.substr(-2);
+  return formattedTime;
+}
+
+/**
+* Does matrix multiplication on specific index of the direction matrix.
+* @param {object} typeOfMatrix matrix to multiply by.
+* @param {int} index  of point to change.
+*/
+function changeMatrix(typeOfMatrix, index) {
+  direction.getMatrixAt(index, instanceMatrix );
+  matrix.multiplyMatrices( instanceMatrix, typeOfMatrix );
+  direction.setMatrixAt( index, matrix );
+  direction.instanceMatrix.needsUpdate = true;
+}
+
+/**
+* Updates GUI with specific point information.
+* @param {int} index index of point.
+*/
+function displayPointValues(index) {
+  time.setValue(formatTime(pose[index].gpsTimestamp));
+  lat.setValue(pose[index].lat);
+  long.setValue(pose[poseStart.start].lng);
+  alt.setValue(pose[index].alt);
+  yaw.setValue(pose[index].yawDeg);
+  roll.setValue(pose[index].rollDeg);
+  pitch.setValue(pose[index].pitchDeg);
+}
+
+/**
+* Hides end of trajectory based on user selected point.
+ */
+function hideOrientation() {
+  if (poseStart.start < poseEnd.end && poseEnd.end < pose.length) {
```

Comment:
You can invert the conditional to reduce nesting.

https://docs.google.com/document/d/16SSi-rIUGxJ5-ksifO6y0InCOwgbRa3lHBLnBJ8Klrk/edit?usp=sharing